model/posts_model.js

```diff
@@ -1,111 +1,40 @@
-        db.raw(`SELECT U.id as user_id, U.username, U.profile_picture_url FROM POSTS as P, USERS as U WHERE P.id = ${id.id} AND P.user_id = U.id`).row((err, row) => {
-            if (err)
-                observer.onError(err);
-            else
-                observer.onNext(row);
-            observer.onCompleted();
-        });
-    });
+    const getUserInfo = db.select('u.id as user_id', 'u.profile_picture_url').from('users as u').join('posts p').on('u.id', 'p.user_id').where({'p.id': id}).row();
 
-    return Rx.Observable.forkJoin(postInfoObservable, userInfoObservable);
-}
+    return Promise.all([getPost, getUserInfo]);
+};
 
 // TODO: add sort
-module.exports.search = (queryParams) => {
-    const {tags, offset = 0, limit = 15, sort, description = ""} = queryParams;
-
+module.exports.search = async ({tags, offset = 0, limit = 15, sort, description = ""}) => {
     let findParams = {};
-    if(tags)
+    if (tags)
         findParams.tags = tags;
-    if(description)
-        findParams.description = {$regex: new RegExp(description, "i")};
+    if (description)
+        findParams.description = description;
 
-    const getPosts = (observer, postIDs) => {
+    const getPosts = async (postIDs) => {
         const hasIDs = postIDs !== undefined;
         const postIDQuery = hasIDs ? ' AND P.id IN $4' : '';
         let params = [`%${description}%`, offset, limit];
         if (hasIDs)
             params.push(postIDs);
 
-        db.raw(`SELECT P.id, P.description, P.image_url, U.id as user_id, U.username FROM POSTS as P, USERS as U WHERE P.USER_ID = U.ID AND P.DESCRIPTION LIKE $1${postIDQuery} ORDER BY P.id DESC OFFSET $2 LIMIT $3`, params).rows((err, rows) => {
-            if (err)
-                observer.onError(err);
-            else
-                observer.onNext(rows);
-            observer.onCompleted();
-        })
+        return db.raw(`SELECT P.id, P.description, P.image_url, U.id as user_id, U.username FROM POSTS as P, USERS as U WHERE P.USER_ID = U.ID AND P.DESCRIPTION LIKE $1${postIDQuery} ORDER BY P.id DESC OFFSET $2 LIMIT $3`, params).rows();
     };
 
-    if (tags !== undefined && tags.length > 0) {
-        const getTagsObservable = Rx.Observable.create(observer => {
-           db.select('pt.post_id').from('post_tags as pt').join('tags t').on('t.id', 'pt.tag_id').where($in('t.name', tags)).rows((err, rows) => {
-               if (err)
-                   observer.onError(err);
-               else
-                   observer.onNext(rows);
-               observer.onCompleted();
-           })
-        });
-        return getTagsObservable.flatMap(postIDs => {
-            return Rx.Observable.create(observer => {
-                getPosts(observer, postIDs);
-            });
-        });
-    }
+    const postIDs = tags !== undefined && tags.length > 0 ? await db.select('pt.post_id').from('post_tags as pt').join('tags t').on('t.id', 'pt.tag_id').where($in('t.name', tags)).rows() : undefined;
```

Comment:
make it multiline?

``` javascript
const postIDs = cond ?
  yes :
  no;
```

Reply:
Yeah, that definitely looks better to do multiline